inference-engine/thirdparty/clDNN/src/batch_to_space.cpp

```diff
@@ -34,88 +34,56 @@ layout batch_to_space_inst::calc_output_layout(batch_to_space_node const& node)
-        }
-        args.push_back(sizes);
-        mem.unlock();
+    if (output_format == format::b_fs_yx_fsv16) {
+        output_format = format::bfyx;
     }
+    auto output_type = desc->output_data_type ? *desc->output_data_type : input_layout.data_type;
+
+    if (node.has_fused_primitives())
+        output_type = node.get_fused_output_layout().data_type;
+
+    const size_t dims_num = format::dimension(input_format);
 
-    const auto& block_shape = args[0];
-    const auto& crops_begin = args[1];
-    const auto& crops_end = args[2];
+    const auto& block_shape = desc->block_shape;
+    const auto& crops_begin = desc->crops_begin;
+    const auto& crops_end = desc->crops_end;
 
-    if (block_shape[0] != 1)
+    if (block_shape.batch[0] != 1)
         CLDNN_ERROR_MESSAGE(node.id(),
             "block_shape[0] is expected to be 1. Actual block_shape[0] is " +
-            std::to_string(block_shape[0]));
+            std::to_string(block_shape.batch[0]));
 
-    if (crops_begin[0] != 0)
+    if (crops_begin.batch[0] != 0)
         CLDNN_ERROR_MESSAGE(node.id(),
             "crops_begin[0] is expected to be 0. Actual crops_begin[0] is " +
-            std::to_string(crops_begin[0]));
+            std::to_string(crops_begin.batch[0]));
 
-    if (crops_end[0] != 0)
+    if (crops_end.batch[0] != 0)
         CLDNN_ERROR_MESSAGE(node.id(),
             "crops_end[0] is expected to be 0. Actual crops_end[0] is " +
-            std::to_string(crops_end[0]));
+            std::to_string(crops_end.batch[0]));
 
-    size_t block_sizes_multiplied = 1;
-    for (const auto& el: block_shape)
-        block_sizes_multiplied *= el;
+    size_t block_sizes_multiplied = block_shape.feature[0];
+    for (size_t i = 0; i < dims_num - 2; ++i)
+        block_sizes_multiplied *= block_shape.spatial[i];
 
     if (input_layout.size.batch[0] % block_sizes_multiplied != 0)
         CLDNN_ERROR_MESSAGE(node.id(),
             "The batch of the input tensor must be divisible by multiplied block sizes = " +
             std::to_string(block_sizes_multiplied));
 
-    for(size_t i = 1; i < dims_num; ++i)
-        if (crops_begin[i] + crops_end[i] >= block_shape[i] * input_layout.size.sizes(input_format)[i])
+    if (crops_begin.feature[0] + crops_end.feature[0] >= block_shape.feature[0] * input_layout.size.sizes(input_format)[1])
             CLDNN_ERROR_MESSAGE(node.id(),
                 "Output dimensions must be positive");
 
-    const auto output_batch = input_layout.size.batch[0] / block_sizes_multiplied;
-
-    std::vector<int32_t> output_shape;
-    output_shape.reserve(dims_num);
-    for (size_t i = 1; i < dims_num; ++i) {
-        output_shape.emplace_back(input_layout.size.sizes(input_format)[i] * block_shape[i] - crops_begin[i] - crops_end[i]);
-    }
+    for (size_t i = 2; i < dims_num; ++i)
+        if (crops_begin.sizes()[i] + crops_end.sizes()[i] >= block_shape.sizes()[i] * input_layout.size.sizes(input_format)[dims_num - i + 1])
```

Comment:
It is better to use `spatial_num(format)` function here too
And instead of `input_layout.size.sizes(input_format)` more suitable would be `input_layout.size.spatial`